src/fsharp/vs/ServiceUntypedParse.fs

```diff
@@ -988,20 +988,12 @@ module UntypedParseImpl =
                 | {TypeName = LongIdentWithDots([x], _)} when x.idText = name -> Some ()
                 | _ -> None
             
-            let rec getKind isClass isInterface isStruct = 
-                function
-                | [] -> isClass, isInterface, isStruct
-                | (SynAttr "Class")::xs -> getKind true isInterface isStruct xs
-                | (SynAttr "AbstractClass")::xs -> getKind true isInterface isStruct xs
-                | (SynAttr "Interface")::xs -> getKind isClass true isStruct xs
-                | (SynAttr "Struct")::xs -> getKind isClass isInterface true xs
-                | _::xs -> getKind isClass isInterface isInterface xs
-
-            match getKind false false false synAttributes with
-            | false, false, false -> Unknown
-            | true, false, false -> Class
-            | false, true, false -> Interface
-            | false, false, true -> Struct
+            match synAttributes with
+            | [ (SynAttr "Class") ]
+            | [ (SynAttr "AbstractClass") ] -> Class
+            | [ (SynAttr "Interface") ] -> Interface
+            | [ (SynAttr "Struct") ] -> Struct
+            | [] -> Unknown
```

Comment:
The original code traverses the entire list recursively, setting one of the three flags as it goes. Your code matches that the list contains exactly one element of one of the expected kinds, otherwise it returns `Unknown`, this is wrong refactoring.

Reply:
I dropped what was unnecessary and also state + asked in my PR message whether cases with multiple occurences of the same attribute can ever happen.

I noticed that I must do a small adjustment, give me a sec.

Reply:
Yes, multiple instances of same attribute are OK, for example `[<TestCase "foo"; TestCase "bar">]`

Reply:
OK, thanks will take it into account